src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/Lightup/ProgrammaticSuppressionInfoWrapper.cs

```diff
@@ -12,21 +12,34 @@ namespace Microsoft.CodeAnalysis.Testing.Lightup
     {
         internal const string WrappedTypeName = "Microsoft.CodeAnalysis.Diagnostics.ProgrammaticSuppressionInfo";
         internal static readonly Type? WrappedType = typeof(Diagnostic).GetTypeInfo().Assembly.GetType(WrappedTypeName);
-        private static readonly Func<object, ImmutableHashSet<(string id, LocalizableString justification)>> s_suppressions;
+        private static readonly Func<object, object> s_suppressions;
 
         private readonly object _instance;
 
         static ProgrammaticSuppressionInfoWrapper()
         {
-            s_suppressions = LightupHelpers.CreatePropertyAccessor<object, ImmutableHashSet<(string id, LocalizableString justification)>>(WrappedType, nameof(Suppressions), ImmutableHashSet<(string id, LocalizableString justification)>.Empty);
+            s_suppressions = LightupHelpers.CreatePropertyAccessor<object, object>(WrappedType, nameof(Suppressions), ImmutableHashSet<(string id, LocalizableString justification)>.Empty);
         }
 
         private ProgrammaticSuppressionInfoWrapper(object instance)
         {
             _instance = instance;
         }
 
-        public ImmutableHashSet<(string id, LocalizableString justification)> Suppressions => s_suppressions(_instance);
+        public ImmutableHashSet<(string id, LocalizableString justification)> Suppressions
+        {
+            get
+            {
+                var suppressions = s_suppressions(_instance);
+                if (suppressions is ImmutableHashSet<(string id, LocalizableString justification)> set)
+                {
+                    return set;
+                }
+
+                // https://github.com/dotnet/roslyn-sdk/issues/1175
+                throw new NotImplementedException("The test framework does not yet support suppressions format from Roslyn 4.10+");
```

Comment:
>NotImplementedException [](http://example.com/codeflow?start=26&length=23)

So, 1175 won't be closed out when this is merged?

Reply:
It will be closed. If in the future this property is needed for some purpose, it will need to be implemented again.